File: 황재연 (Python)/1주차/프로그래머스_추억점수.py
from collections import defaultdict

def solution(name, yearning, photo):
    answer = []
    name_value = defaultdict(int)

    for n,y in zip(name,yearning):
        name_value[n] = y
    
    for cases in photo:
        sums = 0
        for case in cases:
            sums += name_value[case]
        answer.append(sums)

    return answer



name = ["may", "kein", "kain", "radi"]
yearning = [5, 10, 1, 3]
photo = [["may", "kein", "kain", "radi"],["may", "kein", "brin", "deny"], ["kon", "kain", "may", "coni"]]

print(solution(name,yearning,photo))

File: 백성열 (java)/1주차 문제/프로그래머스_추억점수.java
import java.util.HashMap;

public class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String,Integer> map = new HashMap<String,Integer>();
        for(int i = 0 ;i< yearning.length;i++){
            map.put(name[i],yearning[i]);
        }
        for(int i = 0;i<photo.length;i++){
            int result = 0;
            for(int j = 0;j<photo[i].length;j++){
                if(map.get(photo[i][j]) != null){
                    result += map.get(photo[i][j]);
                }
            }
            answer[i] = result;
        }
        return answer;
    }
}